### DISCUSSION TITLE
Chunk time vs number of sample inconsistency.

### DISCUSSION THREAD

**Discussion by Rodolfo-S** [snakers4/silero-vad#211] [Q&A] [answered]
Why are the number of samples for each chunk size different from the time in milliseconds on the same chart?  In the "Performance Metric" page the supported chunk sizes listed are 30, 60, and 100 milliseconds which, at 16kHz sampling rate should be 480, 960, and 1600 samples. However, the number of samples in that same column in parenthesis are listed as 512, 1024, and 1536 which correspond to 32, 64, and 96ms, respectively. 

Now I'm wondering which one is incorrect. Should I be feeding the model 480 samples at a time or 512 samples?

**Comment by snakers4** [comment, marked as answer]
Hi @Rodolfo-S ,

Please use the sample numbers from the docstring:

https://github.com/snakers4/silero-vad/blob/7c671a75c2f82830a550ab4a182e86d3a2609d70/utils_vad.py#L131-L165

> Why are the number of samples for each chunk size different from the time in milliseconds on the same chart? In the "Performance Metric" page the supported chunk sizes listed are 30, 60, and 100 milliseconds which, at 16kHz sampling rate should be 480, 960, and 1600 samples. However, the number of samples in that same column in parenthesis are listed as 512, 1024, and 1536 which correspond to 32, 64, and 96ms, respectively.

We just rounded up for simplicity of presentation.
We used to have more exact numbers of samples, but after model optimization it had to become a multiple of 2**N.
